tox.ini

```diff
@@ -23,10 +23,12 @@ commands =
 [testenv:typing]
 deps =
     {[testenv]deps}
+    types-lxml==2022.4.10
+    types-psutil==5.9.5.4
```

Comment:
Should we make these dependencies of Parsel, optional dependencies, or mention them somewhere in the documentation?

Reply:
Stub packages are only needed when type checking (I even think they are not needed when your code imports parsel but doesn't import psutil or setuptools - I think lxml should be needed because its types are parts of the parsel signatures), I haven't seen any recommendations or practices to put stub packages in reqs of some kind.

Reply:
I would add it as a hard dependency, with an [environment marked for Python 3.8+](https://stackoverflow.com/a/32643122/939364) given [that is the minimum Python version of `types-lxml`](https://github.com/abelcheung/types-lxml/blob/main/pyproject.toml#L10), to ensure the best possible user experience.

If it turns out there are scenarios where this is a problem, we can always remove it in a patch version and document its optional installation instead.

Reply:
Are these packages needed to type check parsel code, or are they needed to type check user code which uses parsel?

Reply:
If it's not required to use typing provided by parsel (excluding .root results - a case where raw lxml data could be exposed), then -1 to add them as dependencies.

Reply:
They are needed to type check parsel itself and shouldn't be needed to type check code that uses parsel. I've copied tests/test_selector.py from this branch to a separate dir, removed all code that imports and uses lxml or setuptools and ran mypy on it, it only asked for parsel itself and ran fine after I installed parsel from this branch.